bpf/bpf_host.c

```diff
@@ -1473,6 +1473,71 @@ int cil_to_netdev(struct __ctx_buff *ctx)
 	if (IS_ERR(ret))
 		goto drop_err;
 
+#if defined(ENABLE_IPV4) && defined(ENABLE_EGRESS_GATEWAY_COMMON)
```

Comment:
The agent doesn't accept the EGW enabled without ipv4. So I think `defined(ENABLE_IPV4)` here is not necessary.

Set `EnableIPv4Masquerade` false here 
https://github.com/cilium/cilium/blob/d43f92323b9f6c0b07b4015d478ded9069aa8c2f/pkg/option/config.go#L2975

Fatal here
https://github.com/cilium/cilium/blob/b7ceaf3a99918afe8473458e6a246065dce46267/pkg/egressgateway/manager.go#L190

Reply:
ack, thank you! Dropped that patch, no need to backport.

I still think we should keep the datapath internally consistent though, for build / complexity testing.

Reply:
Why do we build and test with the unsupported config? Is it a preparatory step for the EGW ipv6 support?